nats-jetstream-http-connector/main.go

```diff
@@ -1,66 +1,101 @@
 package main
 
 import (
-	"context"
 	"io"
 	"log"
 	"net/http"
 	"os"
 	"os/signal"
-	"strings"
 
 	"github.com/nats-io/nats.go"
 	"go.uber.org/zap"
 
 	"github.com/fission/keda-connectors/common"
 )
 
-var (
-	FISSION_CONSUMER = "fission_consumer" // TODO: need to change this
-)
-
 type jetstreamConnector struct {
-	host          string
-	connectordata common.ConnectorMetadata
-	jsContext     nats.JetStreamContext
-	logger        *zap.Logger
+	host            string
+	fissionConsumer string
+	connectordata   common.ConnectorMetadata
+	jsContext       nats.JetStreamContext
+	logger          *zap.Logger
+	consumer        string
+	nc              *nats.Conn
 }
 
-func (conn jetstreamConnector) consumeMessage() {
+func main() {
+
+	host := os.Getenv("NATS_SERVER")
+	consumer := os.Getenv("CONSUMER")
+
+	logger, err := zap.NewProduction()
+	if err != nil {
+		log.Fatalf("can't initialize zap logger: %v", err)
+	}
 
-	sub, err := conn.jsContext.PullSubscribe(os.Getenv("TOPIC"), FISSION_CONSUMER, nats.PullMaxWaiting(512))
+	nc, _ := nats.Connect(host)
+	js, err := nc.JetStream()
 	if err != nil {
-		conn.logger.Fatal("error occurred while consuming message", zap.Error(err))
+		logger.Fatal("error while getting jetstream context:", zap.Error(err))
 	}
-	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
 
+	defer func() {
+		logger.Sync()
+	}()
+
+	connectordata, err := common.ParseConnectorMetadata()
+	if err != nil {
+		logger.Fatal("error occurred while parsing metadata", zap.Error(err))
+	}
+
+	conn := jetstreamConnector{
+		host:            host,
+		fissionConsumer: consumer,
+		connectordata:   connectordata,
+		jsContext:       js,
+		logger:          logger,
+		consumer:        consumer,
+		nc:              nc,
+	}
+	err = conn.consumeMessage()
+	if err != nil {
+		conn.logger.Fatal("error occurred while parsing metadata", zap.Error(err))
+	}
+}
+
+func (conn jetstreamConnector) consumeMessage() error {
+
```

Comment:
can be written in better fashion ?
```
func (conn jetstreamConnector) consumeMessage() {
	// Create durable consumer monitor
	sub, err := conn.jsContext.Subscribe(conn.connectordata.Topic, func(msg *nats.Msg) {
		conn.handleHTTPRequest(msg)
		// Durable is required because if we allow jetstream to create new consumer we
		// will be reading records from the start from the stream.
	}, nats.Durable(conn.consumer), nats.ManualAck())

	ctx, cancel := context.WithCancel(context.Background())
	signalChan := make(chan os.Signal, 2)
	signal.Notify(signalChan, os.Interrupt)
	go func() {
		sig := <-signalChan
		logger.Info("received signal", zap.String("signal", sig.String()))
		cancel()
		<-signalChan
		panic("double signal received")
	}()
	<-ctx.Done()
	logger.Info("unsubscribing and closing connection...")
	err = sub.Unsubscribe()
	if err != nil {
		logger.Error("error while unsubscribing", zap.Error(err))
	}
}
```

Reply:
updated